### PR TITLE
Update to hyper 0.12

### PR DESCRIPTION
**Pull request by astro**
Thanks you for creating this crate!

Please don't forget to publish releases to crates.io even for small-ish changes.

### CODE REVIEW

File: src/use_hyper_client.rs [hunk 1]
```diff
@@ -133,10 +136,10 @@ where
             ],
         );
 
-        let headers = self.headers().iter().fold(headers, |mut acc, header_view| {
-            acc.entry(header_view.name().into())
+        let headers = self.headers().iter().fold(headers, |mut acc, (header_name, header_value)| {
+            acc.entry(header_name.as_str().to_string())
                 .or_insert_with(Vec::new)
-                .push(header_view.value_string());
+                .push(header_value.to_str().unwrap().to_string());
```
Comment (asonix) [inline src/use_hyper_client.rs]: Could you get rid of these unwraps? Most of these functions return `Result<>` anyway, so it shouldn't be too difficult

File: src/use_hyper_client.rs [hunk 2]
```diff
@@ -163,10 +166,10 @@ where
         key: T,
         algorithm: SignatureAlgorithm,
     ) -> Result<&mut Self, Error> {
-        use hyper::header::Authorization;
+        use hyper::header::AUTHORIZATION;
 
         let auth_header = self.authorization_header(key_id, key, algorithm)?;
-        self.headers_mut().set(Authorization(auth_header));
+        self.headers_mut().insert(AUTHORIZATION, HeaderValue::from_str(&auth_header).unwrap());
```
Comment (asonix) [inline src/use_hyper_client.rs]: ditto

File: src/use_hyper_client.rs [hunk 3]
```diff
@@ -178,7 +181,7 @@ where
         algorithm: SignatureAlgorithm,
     ) -> Result<&mut Self, Error> {
         let sig_header = self.signature_header(key_id, key, algorithm)?;
-        self.headers_mut().set_raw("Signature", sig_header);
+        self.headers_mut().insert("Signature", HeaderValue::from_str(&sig_header).unwrap());
```
Comment (asonix) [inline src/use_hyper_client.rs]: here too

File: src/use_hyper_server.rs [hunk 4]
```diff
@@ -22,47 +22,47 @@
...
+fn verify_header<G, B>(req: &Request<B>, header: &str, key_getter: G) -> Result<(), VerificationError>
 where
     G: GetKey,
 {
     let auth_header = SignedHeader::new(header)?;
 
     let headers: Vec<(&str, String)> = req.headers()
         .iter()
-        .map(|header_view| (header_view.name(), header_view.value_string()))
+        .map(|(header_name, header_value)| (header_name.as_str(), header_value.to_str().unwrap().to_string()))
```
Comment (asonix) [inline src/use_hyper_server.rs]: also here

**Review by asonix — CHANGES_REQUESTED** [review]
I pointed out a few places where there are unwraps that should probably not be unwraps.

**Comment by astro** [comment]
Addressed your concerns.

**Review by asonix — APPROVED** [review]

**Comment by asonix** [comment]
Thanks so much!

**Pull request merged**